### DISCUSSION TITLE
what are the metrics to get request per hour and day?

### DISCUSSION THREAD

**Discussion by amolngt** [apache/skywalking#12907] [Q&A] [answered]
hi all,

skywalk is giving metric called service_cpm,  which is giving calls/request per min. i want to do the same for hours and day. didnt find any such metric on official site.

Thank you in advance.

**Comment by wu-sheng** [comment, marked as answer]
If you ask for per hour or per day, we don't have that. When you select the duration larger and the time axis shows in hours/days rather than minutes, it indicates calls per minute in the whole hour.

**Reply by amolngt** [reply]
But it doesn't give actual values. It average out those values.

**Reply by wu-sheng** [reply]
No, it isn't. When you get only 20 calls in one minute, and those are the only calls in the hour and day, the value is
- minute, 20 calls/ min
- hour, 20 calls / 60 mins
- day, 20 calls / 60*24 mins

That is what you should see.

**Reply by amolngt** [reply]
right. but in a day and hour i gets huge number of calls/request. according to service_cpm it gives minutes wise counts correctly. what can i do to get hourly and day wise, is there any custom way to solve this problem ?

**Reply by wu-sheng** [reply]
This seems more like your end bugs or wrong deployment(not set cluster coordinator correctly). In theory, this should be something I showed above, and we are watching our system every day, it works like that.

**Reply by amolngt** [reply]
using service_cpm and hour(20 calls / 60 mins) ....day(20 calls / 60*24 mins) -- skywalk not giving result as expected. **It just dividing the minute wise data**. I am using skywalk node agent, inside my node express application. Using the node agent it sends data to OAP at port 11800. no other configuration i am doing for this. Let me know if i have to do anything. Have a look at below screenshot. for per min i selected last 30 min. for per hour i selected last 1 day and for day wise i selected last 1 week.

![Screenshot from 2024-12-28 23-24-46](https://github.com/user-attachments/assets/b5e9b8c5-6704-4188-8fe4-3dd2b4c7d422)
![Screenshot from 2024-12-28 23-25-06](https://github.com/user-attachments/assets/ba92a62c-e9e1-4faa-8236-87a5d4ab6e4a)
![Screenshot from 2024-12-28 23-25-23](https://github.com/user-attachments/assets/58370491-3ceb-4049-b594-fd3ebe008b4a)


```
const { default: agent } = require('skywalking-backend-js')
  agent.start({
    serviceName: 'APP_NAME',
    collectorAddress: "ip:11800"
  })
```

thank you in advance.

**Reply by wu-sheng** [reply]
You don't need three widgets, all things are done automatically when you use `service_cpm` metric. The values have been calculated in the OAP and will return according to your query duration(step).

**Reply by amolngt** [reply]
But when I change query duration to last week then it ideally show huge number. But currently it showing numbers in points. For every day I have more than 1lac calls.

**Reply by wu-sheng** [reply]
I still don't know what you mean. As I described twice, the values are based on that.

I don't have any more details.
